### PR TITLE
Kick clients that declare an unsupported version

### PR DESCRIPTION
**Pull request by Protonull**

### CODE REVIEW

File: server/src/constants.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export const SUPPORTED_VERSIONS = ['2.0.1-1.18.2']
```
Comment (Gjum) [inline server/src/constants.ts:1]: Is it `2.0.1-1.18.2` or `2.0.1-1.18.2-fabric`?

**Review by Gjum — APPROVED** [review]

**Comment by Huskydog9988** [comment]
I think were all on board with doing the numbered protocol versions, and this seems like a great time to do it since this would be the start of proper protocol enforcement.

File: server/src/constants.ts (hunk 1 above)
Comment (Protonull) [reply server/src/constants.ts:1]: fug, you're right

**Comment by Protonull** [comment]
Going to limit this PR to just being about version enforcement. Amending the protocol itself is a bit outside the scope and begets conversations about other protocol updates.

**Review by Huskydog9988 — APPROVED** [review]

**Pull request merged**